platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/storage/FileSource.java

```diff
@@ -217,6 +261,95 @@ public static String getInternalCachePath(@NonNull Context context) {
     }
   }
 
+  /**
+   * Changes the path of the resources cache database.
+   * Note that the external storage setting needs to be activated in the manifest.
+   *
+   * @param context  the context of the path
+   * @param path     the new database path
+   * @param callback the callback to obtain the result
+   */
+  public static void setResourcesCachePath(@NonNull Context context,
+                                           @NonNull final String path,
+                                           @NonNull final SetResourcesCachePathCallback callback) {
+
+    if (getInstance(context).isActivated()) {
+      final String activatedMessage = "Cannot set path, file source is activated!";
+      Logger.w(TAG, activatedMessage);
+      callback.onError(activatedMessage);
+    } else {
+      final WeakReference<Context> contextWeakReference = new WeakReference<>(context);
+      new Thread(new Runnable() {
+        @Override
+        public void run() {
+          final Context context = contextWeakReference.get();
+          final String message;
+          if (context != null) {
+            if (resourcesCachePath != null) {
+              if (!isExternalStorageConfiguration(context)) {
+                message = "External storage setting is not enabled";
+              } else if (resourcesCachePath.equals(path)) {
+                message = "Changing the path not needed";
+              } else if (!isExternalStorageReadable()) {
+                message = "External storage is not readable";
+              } else if (!isPathWritable(path)) {
+                message = "Path is not writable: " + path;
+              } else {
+                message = null;
+
+                final SharedPreferences.Editor editor =
+                    context.getSharedPreferences(MAPBOX_SHARED_PREFERENCES, Context.MODE_PRIVATE).edit();
+                editor.putString(MAPBOX_SHARED_PREFERENCE_RESOURCES_CACHE_PATH, path).apply();
+
+                new Handler(Looper.getMainLooper()).post(new Runnable() {
+                  @Override
+                  public void run() {
+                    setResourcesCachePath(context, path);
```

Comment:
This can be executed on a background thread.

Reply:
Reinitializing the file source will crash on a background thread.